capa/features/extractors/viv/insn.py

```diff
@@ -181,14 +181,30 @@ def extract_insn_bytes_features(f, bb, insn):
                 yield Bytes(extracted_bytes), insn.va
 
 
+def read_memory(vw, va, size):
+    # as documented in #176, vivisect will not readMemory() when the section is not marked readable.
+    #
+    # but here, we don't care about permissions.
+    # so, copy the viv implementation of readMemory and remove the permissions check.
+    #
+    # this is derived from:
+    #   https://github.com/vivisect/vivisect/blob/5eb4d237bddd4069449a6bc094d332ceed6f9a96/envi/memory.py#L453-L462
+    for mva, mmaxva, mmap, mbytes in vw._map_defs:
+        if va >= mva and va < mmaxva:
+            mva, msize, mperms, mfname = mmap
+            offset = va - mva
+            return mbytes[offset:offset+size]
+    raise envi.SegmentationViolation(va)
+
+
 def read_string(vw, offset):
     try:
         alen = vw.detectString(offset)
```

Comment:
its weird that this routine doesn't care about section permissions, while `readMemory` does